contrib/endpoints/src/api_manager/config.cc

```diff
@@ -398,6 +398,14 @@ bool Config::LoadService(ApiManagerEnvInterface *env,
     string tf;
     ::google::protobuf::TextFormat::PrintToString(service_, &tf);
     env->LogDebug(tf.c_str());
+
+    auto quota_info = service_.quota();
+    for (auto &metric_rule : quota_info.metric_rules()) {
+      for (auto &metric_cost : metric_rule.metric_costs()) {
```

Comment:
not here. you need to create a separate function like

bool Config::LoadQuotaRule(ApiManagerEnvInterface *env) {
  for (auto &rule : service_.quota().metric_rules()) {
    auto method = utils::FindOrNull(method_map_, rule.selector());
    if (method) {
             for (auto &metric_cost : rule.metric_costs()) {
                    method->add_quota_metric(metrick_cost.first, metric_cost.second;
            }
    } else {
       log error
    }
  }
  

    for (auto &rule : service_.quota().quota_rules()) {
          auto method = utils::FindOrNull(method_map_, rule.selector());
           if (method) {
               for (const auto& rule: rule.groups) {
                         // find out the group that rule.group point
                         for (const auto& limit : group.limits()) {
                              method->add_metric(limit.metric, rule.cost);